test/fixtures/valid-env/docker-compose.yml

```diff
@@ -7,16 +7,37 @@ volumes:
     app-env:
 
 services:
+    haproxy:
+        image: dockercloud/haproxy:latest
+        links:
+            - varnish
+        environment:
+            - CERT_FOLDER=/certs
+        volumes:
+            - /var/run/docker.sock:/var/run/docker.sock
+            - .docker/certs:/certs
+        ports:
+            - "80:80"
+            - "443:443"
+
+    varnish:
+        container_name: m2-varnish
+        image: wearejh/magento-varnish:latest
+        environment:
+            - FORCE_SSL=yes
+        depends_on:
+            - nginx
+
     nginx:
         container_name: m2
         image: nginx:stable-alpine
+        working_dir: /var/www
         volumes:
-            - app-pub:/var/www/pub
             - .docker/nginx/sites:/etc/nginx/conf.d
-        working_dir: /var/www
-        ports:
-            - "80:80"
-            - "443:443"
+        volumes_from:
```

Comment:
Compose file v2 is far from deprecated, but I wonder if we should consider stepping up to v3 and using named volumes rather than taking both anonymous and named volumes with `volumes_from`

Reply:
That's the plan, going to look at how well this works in dev-playground ... Also got to make sure those no weird behaviour from restoring state on future starts.